uniref-rest/src/main/java/org/uniprot/api/uniref/repository/store/UniRefEntryStoreRepository.java

```diff
@@ -0,0 +1,151 @@
+package org.uniprot.api.uniref.repository.store;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+
+import lombok.extern.slf4j.Slf4j;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+
+import org.springframework.stereotype.Component;
+import org.uniprot.api.common.exception.ResourceNotFoundException;
+import org.uniprot.api.common.repository.search.page.impl.CursorPage;
+import org.uniprot.api.common.repository.store.BatchStoreIterable;
+import org.uniprot.api.rest.request.SearchRequest;
+import org.uniprot.api.uniref.request.UniRefIdRequest;
+import org.uniprot.api.uniref.service.UniRefEntryResult;
+import org.uniprot.core.uniref.*;
+import org.uniprot.core.uniref.impl.AbstractUniRefMemberBuilder;
+import org.uniprot.core.uniref.impl.RepresentativeMemberBuilder;
+import org.uniprot.core.uniref.impl.UniRefEntryBuilder;
+import org.uniprot.core.uniref.impl.UniRefMemberBuilder;
+
+/**
+ * @author jluo
+ * @date: 20 Aug 2019
+ */
+@Component
+@Slf4j
+public class UniRefEntryStoreRepository {
+
+    private static final String CLUSTER_ID_NOT_FOUND =
+            "Unable to get UniRefEntry from store. ClusterId: ";
+    private final RetryPolicy<Object> retryPolicy =
+            new RetryPolicy<>()
+                    .handle(IOException.class)
+                    .withDelay(Duration.ofMillis(100))
+                    .withMaxRetries(5);
+    private final UniRefMemberStoreClient unirefMemberStore;
+    private final UniRefLightStoreClient uniRefLightStore;
+
+    public UniRefEntryStoreRepository(
+            UniRefMemberStoreClient unirefMemberStore, UniRefLightStoreClient uniRefLightStore) {
+        this.unirefMemberStore = unirefMemberStore;
+        this.uniRefLightStore = uniRefLightStore;
+    }
+
+    public UniRefEntryResult getEntryById(String idValue, UniRefIdRequest uniRefIdRequest) {
+        UniRefEntryResult.UniRefEntryResultBuilder builder = UniRefEntryResult.builder();
+        UniRefEntryLight entryLight = getUniRefEntryLightFromStore(idValue);
+        List<String> pageMemberIds = entryLight.getMembers();
+        boolean convertCommon = true;
+        if (!uniRefIdRequest.isComplete()) {
+            CursorPage page = getPage(uniRefIdRequest, entryLight.getMemberCount());
+            builder.page(page);
+            pageMemberIds = getMembersPage(page, entryLight);
+            convertCommon = (page.getOffset() == 0);
+        }
+        builder.entry(buildUniRefEntry(entryLight, pageMemberIds, convertCommon));
+        return builder.build();
+    }
+
+    private UniRefEntry buildUniRefEntry(
+            UniRefEntryLight entryLight, List<String> pageMemberIds, boolean convertCommon) {
+        UniRefEntryBuilder builder = new UniRefEntryBuilder();
+
```

Comment:
Should we return id of unireflight with all member pages?

Reply:
I prefer "pages" as clean as possible.  why do you think it would be useful?

Reply:
I was just saying that so that we can relate the members to the clusterId if we are in the middle of the response in multiple paginated calls. May be needed for troubleshooting or something. I am not sure.